clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -1832,12 +1845,11 @@ raw2trace_t::adjust_and_emit_rseq_buffer(raw2trace_thread_data_t *tdata, addr_t
         // up on distinguishing a side exit whose target is the end PC fall-through from
         // a completion.
         log(4, "Rseq completed normally\n");
-    }
-    // We actually won't see the handler PC as the rseq abort event has the
-    // commit PC and we have it call here and don't wait for a subsequent
-    // instr (or signal interruption) to find the handler PC; so here it's
-    // not really the "next_pc" but the "exit_pc".
-    else if (next_pc == tdata->rseq_commit_pc_) {
+    } else if ((at_abort_marker && next_pc == tdata->rseq_handler_pc_) ||
+               // Old traces have the commit PC instead of the handler in the abort and
+               // signal events (so not really the "next_pc" but the "exit_pc") and
+               // don't have abort events on a signal aborting an rseq region.
+               next_pc == tdata->rseq_commit_pc_) {
```

Comment:
Add `&& !rseq_ever_saw_entry_` to the second one?

Reply:
The problem is there's a window between PR #5967 and this one: I guess it's very narrow since it looks like this will be merged right after but at the time I wasn't sure.  Maybe we can not support that PR without this one: I might have to update a unit test for that.  Trying it.

Reply:
Done.